tests/core/job_test.py

```diff
@@ -241,59 +251,53 @@ def test__eq__false(self):
 class JobSchedulerTestCase(TestCase):
     @setup
     def setup_job(self):
-        self.scheduler = Turtle()
-        run_collection = Turtle()
-        node_pool = Turtle()
-        self.job = job.Job(
-            "jobname",
-            self.scheduler,
-            run_collection=run_collection,
-            node_pool=node_pool,
-        )
-        self.job_scheduler = job.JobScheduler(self.job)
+        self.scheduler = scheduler.ConstantScheduler()
+        run_collection = jobrun.JobRunCollection(20)
+        mock_graph = mock.Mock(autospec=True)
 
-    def test_restore_job_state(self):
-        run_collection = mocks.MockJobRunCollection(
-            get_scheduled=lambda: ['a'],
+        mock_graph.get_action_map.return_value = {}
+        node_pool = node.NodePool.from_node(
+            node=mock.Mock(name="foo", autospec=True)
         )
-        self.job_scheduler.job = Turtle(runs=run_collection)
-        self.job_scheduler._set_callback = Turtle()
-        state_data = 'state_data_token'
-        self.job_scheduler.restore_state(state_data)
-        assert_call(self.job_scheduler.job.restore_state, 0, state_data)
-        assert_length(self.job_scheduler._set_callback.calls, 1)
-        assert_call(self.job_scheduler._set_callback, 0, 'a')
+
+        mock_graph.action_map = {}
+        self.job = mock.Mock(autospec=True)
```

Comment:
I'm not sure if autospec=True does anything when you're just creating a mock (as opposed to using mock.patch). spec_set might do what you want?